letsencrypt/client/revoker.py

```diff
@@ -6,146 +14,513 @@
+        is installed
+
+    """
+    PathStatus = collections.namedtuple("PathStatus", "path status")
+    """Convenience container to hold path and status info"""
+
+    DELETED_MSG = "This file has been moved or deleted"
+    CHANGED_MSG = "This file has changed"
+
+    def __init__(self, cert_path):
+        """Cert initialization
+
+        :param str cert_filepath: Name of file containing certificate in
+            PEM format.
+
+        """
+        try:
+            self.cert = M2Crypto.X509.load_cert(cert_path)
+        except (IOError, M2Crypto.X509.X509Error):
+            raise errors.LetsEncryptRevokerError(
+                "Error loading certificate: %s" % cert_path)
+
+        self.idx = -1
+
+        self.orig = None
+        self.orig_key = None
+        self.backup_path = ""
+        self.backup_key_path = ""
+
+        self.installed = ["Unknown"]
+
+    @classmethod
+    def fromrow(cls, row, backup_dir):
+        # pylint: disable=protected-access
+        """Initialize Cert from a csv row."""
+        idx = int(row[0])
+        backup = Revoker._get_backup(backup_dir, idx, row[1])
+        backup_key = Revoker._get_backup(backup_dir, idx, row[2])
+
+        obj = cls(backup)
+        obj.add_meta(idx, row[1], row[2], backup, backup_key)
+        return obj
+
+    def get_row(self):
+        """Returns a list in CSV format. If meta data is available."""
+        if self.orig is not None and self.orig_key is not None:
+            return [str(self.idx), self.orig.path, self.orig_key.path]
+        return None
+
+    def add_meta(self, idx, orig, orig_key, backup, backup_key):
+        """Add meta data to cert
+
+        :param int idx: convenience index for revoker
+        :param tuple orig: (`str` original certificate filepath, `str` status)
+        :param tuple orig_key: (`str` original auth key path, `str` status)
+        :param str backup: backup certificate filepath
+        :param str backup_key: backup key filepath
+
+        """
+        status = ""
+        key_status = ""
+
+        # Verify original cert path
+        if not os.path.isfile(orig):
+            status = Cert.DELETED_MSG
+        else:
+            o_cert = M2Crypto.X509.load_cert(orig)
+            if self.get_fingerprint() != o_cert.get_fingerprint(md="sha1"):
+                status = Cert.CHANGED_MSG
+
+        # Verify original key path
+        if not os.path.isfile(orig_key):
+            key_status = Cert.DELETED_MSG
+        else:
+            with open(orig_key, "r") as fd:
+                key_pem = fd.read()
+            with open(backup_key, "r") as fd:
+                backup_key_pem = fd.read()
+            if key_pem != backup_key_pem:
+                key_status = Cert.CHANGED_MSG
+
+        self.idx = idx
+        self.orig = Cert.PathStatus(orig, status)
+        self.orig_key = Cert.PathStatus(orig_key, key_status)
+        self.backup_path = backup
+        self.backup_key_path = backup_key
+
+    def get_installed_msg(self):
```

Comment:
https://github.com/letsencrypt/lets-encrypt-preview/blob/revoker/letsencrypt/client/display/revocation.py#L45

There is a question about how you want to display these and what questions are pertinent to the user.  Is it worse off to have too many functions available or contain half of an interface?  I guess that since this is so tightly coupled with the Revoker that removing the unnecessary public functions would be fine and we can add them as we need them?